### ISSUE TITLE
Add support for unicode literals

### ISSUE THREAD

**Issue by lonelyenvoy** [solidity-parser/parser#38]
Since v0.7.0, Solidity has a new syntax for unicode literals (see [documents](https://docs.soliditylang.org/en/v0.8.1/types.html#unicode-literals)):

```solidity
string memory a = unicode"Hello 😃";
```
However, this is not currently supported by the parser. 

For the following Javascript code:
```js
const parser = require('@solidity-parser/parser');
const code = `
  contract test {
    function f() {
      string memory a = unicode"Hello 😃";
    }
  }
`;
parser.parse(code);
```

An error will be thrown:
```
/.../node_modules/@solidity-parser/parser/src/index.ts:72
    throw new ParserError({ errors: listener.getErrors() })
          ^
Error: extraneous input '"Hello 😃"' expecting ';' (4:31)
    at Object.parse (/.../node_modules/@solidity-parser/parser/src/index.ts:72:11)
```

**Comment by fvictorio** [comment]
Thanks @lonelyenvoy, I didn't know about this feature!

**Comment by fvictorio** [comment]
Fixed in `0.12.0`
